raster/r.mfilter/testsuite/test_r_mfilter.py

```diff
@@ -0,0 +1,434 @@
+        return f
+
+    @classmethod
+    def setUpClass(cls):
+        cls.use_temp_region()
+        cls.runModule("g.region", raster="elevation")
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.del_temp_region()
+        if cls.to_remove:
+            cls.runModule(
+                "g.remove", flags="f", type="raster", name=",".join(cls.to_remove)
+            )
+
+    def test_sequential(self):
+        """Test output with sequential filter type."""
+        test_case = "test_sequential"
+        output = "{}_raster".format(test_case)
+        output_threaded = "{}_threaded_raster".format(test_case)
+        self.to_remove.extend([output, output_threaded])
+
+        filter = self.create_filter(self.filter_options["sequential"])
+        self.assertModule(
+            "r.mfilter",
+            input="elevation",
+            output=output,
+            filter=filter.name,
+        )
+        self.assertModule(
+            "r.mfilter",
+            input="elevation",
+            output=output_threaded,
+            filter=filter.name,
+            nprocs=4,
+        )
+        filter.close()
+        self.assertRasterFitsUnivar(
+            raster=output,
+            reference=self.test_results[test_case],
+            precision=1e-5,
+        )
+        self.assertRasterFitsUnivar(
+            raster=output_threaded,
+            reference=self.test_results[test_case],
+            precision=1e-5,
+        )
+
+    def test_parallel(self):
+        """Test output with parallel filter type."""
+        test_case = "test_parallel"
+        output = "{}_raster".format(test_case)
+        output_threaded = "{}_threaded_raster".format(test_case)
+        self.to_remove.extend([output, output_threaded])
+
+        filter = self.create_filter(self.filter_options["parallel"])
+        self.assertModule(
+            "r.mfilter",
+            input="elevation",
+            output=output,
+            filter=filter.name,
+        )
+        self.assertModule(
+            "r.mfilter",
+            input="elevation",
+            output=output_threaded,
+            filter=filter.name,
+            nprocs=4,
+        )
+        filter.close()
+        self.assertRasterFitsUnivar(
+            raster=output,
+            reference=self.test_results[test_case],
+            precision=1e-5,
+        )
+        self.assertRasterFitsUnivar(
+            raster=output_threaded,
+            reference=self.test_results[test_case],
+            precision=1e-5,
+        )
+
+    def test_sequential_null(self):
+        """Test output with sequential filter type with null mode enabled."""
+        test_case = "test_sequential_null"
+        output = "{}_raster".format(test_case)
+        output_z = "{}_z_raster".format(test_case)
+        self.to_remove.extend([output, output_z])
+
+        filter = self.create_filter(self.filter_options["sequential"])
+        self.assertModule(
+            "r.mfilter",
+            input="slope",
```

Comment:
My version of nc_basic_spm_grass7 location does not have a slope map. Can it be replaced with something else to allow testing on a basic location?

Reply:
@aaronsms Slope is missing in [the basic location](https://grass.osgeo.org/sampledata/north_carolina/nc_basic_spm_grass7.zip). It can be a good idea to use the basic location for writing test scripts.

@marisn Any specific reasons why you don't use the full nc_spm_08_grass7?

Reply:
I will replace 'slope' with 'lakes'.

Reply:
Why not `elevation`?